### PR TITLE
doc: release-notes: mfd and regulator release notes

### PR DESCRIPTION
**Pull request by aasinclair**
Added release notes for mfd and regulators

### CODE REVIEW

**Review by gmarull — DISMISSED** [review]

File: doc/releases/release-notes-3.7.rst [hunk 1]
```diff
@@ -656,6 +666,20 @@ Drivers and Sensors
 
 * Regulators
 
+  * New driver :dtcompatible:`cirrus,cp9314`.
+  * Added ``regulator-boot-off`` property to common regulator driver.
+    Updated :dtcompatible:`adi,adp5360-regulator`, :dtcompatible:`nordic,npm1300-regulator`,
+    :dtcompatible:`nordic,npm6001-regulator` and :dtcompatible:`x-powers,axp192-regulator`.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:672]:
```suggestion
    :dtcompatible:`nordic,npm6001-regulator` and :dtcompatible:`x-powers,axp192-regulator`
```

File: doc/releases/release-notes-3.7.rst [hunk 2]
```diff
@@ -656,6 +666,20 @@ Drivers and Sensors
...
+    to use new property.
```
Comment (aescolar) [inline doc/releases/release-notes-3.7.rst:673]:
```suggestion
    to use this new property.
```

**Review by aescolar — COMMENTED** [review]
Thanks, just 2 minors

**Review by aescolar — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

File: doc/releases/release-notes-3.7.rst [hunk 3]
```diff
@@ -656,6 +666,20 @@ Drivers and Sensors
...
+  * Added ``regulator-boot-off`` property to common regulator driver.
+    Updated :dtcompatible:`adi,adp5360-regulator`, :dtcompatible:`nordic,npm1300-regulator`,
+    :dtcompatible:`nordic,npm6001-regulator` and :dtcompatible:`x-powers,axp192-regulator`
+    to use this new property.
+  * Added power management for :dtcompatible:`renesas,smartbond-regulator`.
+  * Added ``is_enabled`` shell command.
+  * Removed use of busy wait for single threaded systems.
+  * Fixed control of DCDC2 output for :dtcompatible:`x-powers,axp192-regulator`.
+  * Fixed current and voltage get functions for :dtcompatible:`renesas,smartbond-regulator`.
+  * Fixed NXP VEF Kconfig leakage.
```
Comment (kartben) [inline doc/releases/release-notes-3.7.rst:679]:
```suggestion
  * Fixed NXP VREF Kconfig leakage.
```

**Review by kartben — CHANGES_REQUESTED** [review]
minor typo - thx

**Review by aescolar — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**